src/sentry/testutils/pytest/json_report_reruns.py

```diff
@@ -0,0 +1,46 @@
+"""Extend the pytest-json-report plugin with a new "reruns" attribute.
+
+Each test will (if it's run more than once) store its prior results in a
+`reruns` array attribute, where each item may have `setup` `call` and
+`teardown` sections.
+
+See also:
+    * https://github.com/pytest-dev/pytest-rerunfailures
+    * https://github.com/numirias/pytest-json-report
+    * https://getsentry.atlassian.net/browse/DEVINFRA-630
+"""
+
+import pytest
+import pytest_jsonreport.plugin
+
+JSONTestItem = dict[str, list[object]]
+JSONTestItems = dict[str, JSONTestItem]
```

Comment:
personally I try and use aliases exceedingly sparingly -- especially if they're only used once.  and even then I'll explicitly mark them with `TypeAlias` so the type checker doesn't accidentally treat them as normal variables (or the `type` statement which we can use now).  but in this case I'd probably just inline this -- especially because there's only one place it gets used